### DISCUSSION TITLE
GitHub extension for VS Code and PR from forks of a private repository

### DISCUSSION THREAD

**Discussion by giact** [community/community#35703] [Codespaces] [answered]
Hello community!

I have a private repository.
If I add collaborators to it and they fork the private repo and then send pull requests, how am I able to work on them from a Codespace using the GitHub extension for VS Code?

The extension correctly sees the PR, but I cannot check them out because the Codespace token lacks permission to read from the collaborator's repository (even if my GitHub user can).

According to https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces there used to be an option to allow such access, but it has been deprecated.
Instead it says I should use the `devcontainer.json` file (which I have not tested yet) but then I have two issues with it:

1. I would have to "pollute" the source code with information that purely relates to GitHub
2. Most importantly, I would have to modify and commit that file every time a new collaborator wants to send a PR, and I think that's rather absurd 🤔 

Is there a way to have a Codespace gain automatic access to the repositories that my GitHub user can already access and which have opened pull requests towards my repo?

Thanks in advance for any help! ❤️

**Comment by legomushroom** [comment, marked as answer]
@giact the token we create for a Codespace is indeed scoped only to that repo for security reasons. I don't think there is a good solution for this atm, but you can try:

- pull in the branch from the fork back to your repo and check it out
- add more powerful token by hand. e.g.
  - generate a token on the developer settings page in your profile
  - add it as a secret in repo Codespaces settings
  - overwrite `$GITHUB_TOKEN` environment variable inside the Codespace with the secret during Codespace creation process
  This, of course, defies the security safeguards we put into the system so I'd rather go the previous route, just want to point out the possibility

I will proxy this feedback to the team, but since there is some inter-team coordination is required (GitHub PR extension is maintained by a different team), I don't expect this to be resolved particularly quickly. Hope that the ideas above can unblock you meanwhile ⏲️

**Reply by giact** [reply]
@legomushroom thanks a lot for the reply!

I was already using the first method but it was cumbersome.

Using your idea, I temporarily bypassed the issue this way:

- I created a personal access token with all repo permissions
- Added that token in a Codespaces secret called `PRIVATE_REPOS_TOKEN` and enabled it only on the necessary Codespace
- Created my own credential helper wrapper that replaces the password with the new token:
  `echo -e '#!/bin/sh\n/.codespaces/bin/gitcredential_github.sh | grep -v ^password\necho password=$PRIVATE_REPOS_TOKEN' > ~/.local/bin/gitcredential_github.sh && chmod +x ~/.local/bin/gitcredential_github.sh`
- Reconfigured Git to use that credential helper:
  `sudo git config --system credential.helper ~/.local/bin/gitcredential_github.sh`

Now everything seems to work fine.

I will remove this workaround once the GitHub devs have fixed it.

Thanks again for your help! ❤️ 

***

Edit: improved the workaround by using `credential.helper`
